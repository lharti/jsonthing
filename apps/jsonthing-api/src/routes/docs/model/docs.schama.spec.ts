import { DocSchema, docsModel } from './doc.schema'

// eslint-disable-next-line @typescript-eslint/no-require-imports
const mockingoose = require('mockingoose')

describe('docSchema', () => {
    beforeEach(() => {
        mockingoose.resetAll()
    })

    it('should match snapshot', () => {
        expect.assertions(1)

        expect(DocSchema).toMatchSnapshot()
    })

    it('should define schema object', () => {
        expect.assertions(1)

        expect(DocSchema.obj).toMatchInlineSnapshot(`
            {
              "content": {
                "required": true,
                "type": [Function],
              },
              "title": {
                "minlength": 1,
                "required": true,
                "type": [Function],
              },
            }
        `)
    })

    it('should have the right options', () => {
        expect.assertions(1)

        // @ts-expect-error: exists, but not defined in mongoose types
        expect(DocSchema.options).toMatchInlineSnapshot(`
            {
              "_id": true,
              "autoIndex": null,
              "bufferCommands": true,
              "capped": false,
              "discriminatorKey": "__t",
              "id": true,
              "minimize": true,
              "optimisticConcurrency": false,
              "pluralization": true,
              "read": null,
              "shardKey": null,
              "strict": true,
              "strictQuery": false,
              "toJSON": {
                "transform": [Function],
                "virtuals": true,
              },
              "toObject": {
                "transform": [Function],
                "virtuals": true,
              },
              "typeKey": "type",
              "validateBeforeSave": true,
              "validateModifiedOnly": false,
              "versionKey": "__v",
            }
        `)
    })

    describe('toObject', () => {
        it('should return pretty doc', async () => {
            expect.assertions(1)

            const random = Math.random().toString()

            const doc = await docsModel.create({
                title: random,
                content: { test: random },
            })

            expect(doc.toJSON()).toStrictEqual({
                id: doc._id.toString(),

                content: { test: random },
                title: random,
            })
        })
    })

    describe('toJSON', () => {
        it('should return pretty doc', async () => {
            expect.assertions(1)

            const random = Math.random().toString()

            const doc = await docsModel.create({
                title: random,
                content: { test: random },
            })

            expect(doc.toJSON()).toStrictEqual({
                id: doc._id.toString(),

                content: { test: random },
                title: random,
            })
        })
    })
})
