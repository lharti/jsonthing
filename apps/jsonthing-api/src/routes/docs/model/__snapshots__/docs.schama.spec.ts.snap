// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docSchema should match snapshot 1`] = `
Schema {
  "$globalPluginsApplied": true,
  "$id": 1,
  "_indexes": [],
  "_searchIndexes": [],
  "_userProvidedOptions": {
    "id": true,
    "toJSON": {
      "transform": [Function],
      "virtuals": true,
    },
    "toObject": {
      "transform": [Function],
      "virtuals": true,
    },
  },
  "aliases": {},
  "callQueue": [],
  "childSchemas": [],
  "inherits": {},
  "mapPaths": [],
  "methodOptions": {},
  "methods": {},
  "nested": {},
  "obj": {
    "content": {
      "required": true,
      "type": [Function],
    },
    "title": {
      "minlength": 1,
      "required": true,
      "type": [Function],
    },
  },
  "options": {
    "_id": true,
    "autoIndex": null,
    "bufferCommands": true,
    "capped": false,
    "discriminatorKey": "__t",
    "id": true,
    "minimize": true,
    "optimisticConcurrency": false,
    "pluralization": true,
    "read": null,
    "shardKey": null,
    "strict": true,
    "strictQuery": false,
    "toJSON": {
      "transform": [Function],
      "virtuals": true,
    },
    "toObject": {
      "transform": [Function],
      "virtuals": true,
    },
    "typeKey": "type",
    "validateBeforeSave": true,
    "validateModifiedOnly": false,
    "versionKey": "__v",
  },
  "paths": {
    "__v": SchemaNumber {
      "_index": null,
      "_presplitPath": [
        "__v",
      ],
      "getters": [],
      "instance": "Number",
      "options": SchemaNumberOptions {
        "type": [Function],
      },
      "path": "__v",
      "setters": [],
      "validators": [],
      Symbol(mongoose#schemaType): true,
    },
    "_id": SchemaObjectId {
      "_index": null,
      "_presplitPath": [
        "_id",
      ],
      "defaultValue": [Function],
      "getters": [],
      "instance": "ObjectId",
      "options": SchemaObjectIdOptions {
        "auto": true,
        "type": "ObjectId",
      },
      "path": "_id",
      "setters": [
        [Function],
      ],
      "validators": [],
      Symbol(mongoose#schemaType): true,
    },
    "content": SchemaMixed {
      "_index": null,
      "_presplitPath": [
        "content",
      ],
      "getters": [],
      "instance": "Mixed",
      "isRequired": true,
      "options": SchemaTypeOptions {
        "required": true,
        "type": [Function],
      },
      "originalRequiredValue": true,
      "path": "content",
      "requiredValidator": [Function],
      "setters": [],
      "validators": [
        {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
      Symbol(mongoose:schema_mixed): true,
    },
    "title": SchemaString {
      "_index": null,
      "_presplitPath": [
        "title",
      ],
      "enumValues": [],
      "getters": [],
      "instance": "String",
      "isRequired": true,
      "minlengthValidator": [Function],
      "options": SchemaStringOptions {
        "minlength": 1,
        "required": true,
        "type": [Function],
      },
      "originalRequiredValue": true,
      "path": "title",
      "regExp": null,
      "requiredValidator": [Function],
      "setters": [],
      "validators": [
        {
          "message": "Path \`{PATH}\` is required.",
          "type": "required",
          "validator": [Function],
        },
        {
          "message": "Path \`{PATH}\` (\`{VALUE}\`) is shorter than the minimum allowed length (1).",
          "minlength": 1,
          "type": "minlength",
          "validator": [Function],
        },
      ],
      Symbol(mongoose#schemaType): true,
    },
  },
  "plugins": [
    {
      "fn": [Function],
      "opts": {
        "deduplicate": true,
      },
    },
    {
      "fn": [Function],
      "opts": {
        "deduplicate": true,
      },
    },
    {
      "fn": [Function],
      "opts": {
        "deduplicate": true,
      },
    },
    {
      "fn": [Function],
      "opts": {
        "deduplicate": true,
      },
    },
    {
      "fn": [Function],
      "opts": {
        "deduplicate": true,
      },
    },
  ],
  "query": {},
  "s": {
    "hooks": Kareem {
      "_posts": Map {
        "save" => [
          {
            "fn": [Function],
          },
          {
            "fn": [Function],
          },
          {
            "fn": [Function],
          },
        ],
        "init" => [
          {
            "fn": [Function],
          },
        ],
      },
      "_pres": Map {
        "save" => [
          {
            "fn": [Function],
            "isAsync": false,
          },
          {
            "fn": [Function],
            "isAsync": false,
          },
          {
            "fn": [Function],
            "isAsync": false,
          },
          {
            "fn": [Function],
            "isAsync": false,
          },
        ],
        "remove" => [
          {
            "fn": [Function],
            "isAsync": false,
          },
        ],
      },
    },
  },
  "singleNestedPaths": {},
  "statics": {},
  "subpaths": {},
  "tree": {
    "__v": [Function],
    "_id": {
      "auto": true,
      "type": "ObjectId",
    },
    "content": {
      "required": true,
      "type": [Function],
    },
    "id": VirtualType {
      "getters": [
        [Function],
      ],
      "options": {},
      "path": "id",
      "setters": [],
    },
    "title": {
      "minlength": 1,
      "required": true,
      "type": [Function],
    },
  },
  "virtuals": {
    "id": VirtualType {
      "getters": [
        [Function],
      ],
      "options": {},
      "path": "id",
      "setters": [],
    },
  },
}
`;
